--- nodelint-devel/Makefile	2012-10-10 19:54:40.000000000 -0700
+++ nodelint-devel/Makefile	2012-10-10 19:54:45.000000000 -0700
@@ -30,11 +30,11 @@
 	perl -pi -e 's{path\.join\(SCRIPT_DIRECTORY, '\''jslint/jslint\.js'\''\)}{"$(PACKAGEDATADIR)/jslint.js"}' $(BUILDDIR)/nodelint
 
 install: build doc
-	install --directory $(PACKAGEDATADIR)
-	install --mode 0644 $(BUILDDIR)/jslint.js $(PACKAGEDATADIR)/jslint.js
-	install --mode 0644 $(BUILDDIR)/config.js $(ETCDIR)/nodelint.conf
-	install --mode 0755 $(BUILDDIR)/nodelint $(BINDIR)/nodelint
-	install --directory $(MANDIR)/man1/
+	install -d $(PACKAGEDATADIR)
+	install -m 0644 $(BUILDDIR)/jslint.js $(PACKAGEDATADIR)/jslint.js
+	install -m 0644 $(BUILDDIR)/config.js $(ETCDIR)/nodelint.conf
+	install -m 0755 $(BUILDDIR)/nodelint $(BINDIR)/nodelint
+	install -d $(MANDIR)/man1/
 	cp -a man1/nodelint.1 $(MANDIR)/man1/
 
 uninstall:
