--- busybox-1.19.0/findutils/find.c
+++ busybox-1.19.0-find/findutils/find.c
@@ -932,7 +932,10 @@ static action*** parse_params(char **arg
  * expression is reached.
  */
 		/* Options */
-		if (0) { }
+		if (parm == OPT_FOLLOW) {
+                        dbg("follow enabled: %d", __LINE__);
+                        G.recurse_flags |= ACTION_FOLLOWLINKS | ACTION_DANGLING_OK;
+		}
 #if ENABLE_FEATURE_FIND_XDEV
 		else if (parm == OPT_XDEV) {
 			dbg("%d", __LINE__);
