diff -urN libxslt-1.1.28-prev/doc/Makefile.am libxslt-1.1.28/doc/Makefile.am
--- libxslt-1.1.28-prev/doc/Makefile.am	2013-03-22 08:50:32.262000238 -0700
+++ libxslt-1.1.28/doc/Makefile.am	2013-03-22 08:51:30.019998872 -0700
@@ -240,7 +240,7 @@
 	($(XSLTPROC) -o ../libxslt/libxslt.syms $(srcdir)/syms.xsl $(srcdir)/symbols.xml)
 
 xsltproc.1: $(srcdir)/xsltproc.xml
-	-@$(XSLTPROC) --nonet $(srcdir)/xsltproc.xml
+	-@$(XSLTPROC) $(srcdir)/xsltproc.xml
 
 # Note that in the following, xmllint output is piped to xsltproc
 search.php: $(srcdir)/api.xsl $(srcdir)/site.xsl $(srcdir)/search.templ \
diff -urN libxslt-1.1.28-prev/doc/Makefile.in libxslt-1.1.28/doc/Makefile.in
--- libxslt-1.1.28-prev/doc/Makefile.in	2013-03-22 08:50:32.274000027 -0700
+++ libxslt-1.1.28/doc/Makefile.in	2013-03-22 08:51:36.677000033 -0700
@@ -787,7 +787,7 @@
 	($(XSLTPROC) -o ../libxslt/libxslt.syms $(srcdir)/syms.xsl $(srcdir)/symbols.xml)
 
 xsltproc.1: $(srcdir)/xsltproc.xml
-	-@$(XSLTPROC) --nonet $(srcdir)/xsltproc.xml
+	-@$(XSLTPROC) $(srcdir)/xsltproc.xml
 
 # Note that in the following, xmllint output is piped to xsltproc
 search.php: $(srcdir)/api.xsl $(srcdir)/site.xsl $(srcdir)/search.templ \
