--- erts/emulator/Makefile.in~	2011-05-19 13:16:04.000000000 -0700
+++ erts/emulator/Makefile.in	2011-05-19 13:17:02.000000000 -0700
@@ -572,7 +572,7 @@
 # Usually the same as the default rule, but certain platforms (i.e. win32) mix
 # different compilers
 $(OBJDIR)/beam_emu.o: beam/beam_emu.c
-	$(EMU_CC) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) $(INCLUDES) -c $< -o $@
+	$(EMU_CC) $(INCLUDES) $(subst -O2, $(GEN_OPT_FLGS), $(CFLAGS)) -c $< -o $@
 
 $(OBJDIR)/dll_sys.o: sys/$(ERLANG_OSTYPE)/sys.c
 	$(CC) $(CFLAGS) -DERL_RUN_SHARED_LIB=1 $(INCLUDES) -c $< -o $@
