diff -ur garden_maker/update.sh garden_maker/update.sh
--- garden_maker/update.sh	2013-11-18 19:07:25.627436693 -0800
+++ garden_maker/update.sh	2013-11-18 19:07:48.105431936 -0800
@@ -1,4 +1,10 @@
-#!/bin/bash
+#!/bin/sh
+
+echo "Downloading dashboard database for CouchDB:" >&2 &&
+curl 'http://staging.dev.medicmobile.org/downloads/dashboard-medic.couch' \
+  -# -o assets/dashboard.couch &&
+\
+echo "Downloading Futon assets for CouchDB:" >&2 &&
+curl 'https://codeload.github.com/garden20/garden-futon/zip/master' \
+  -# -o assets/futon.zip
 
-curl   http://reupholster.iriscouch.com/dashboard/_design/dashboard/static/dashboard.couch -o assets/dashboard.couch
-wget   https://github.com/garden20/garden-futon/archive/master.zip -O assets/futon.zip
