diff -ur open-vm-tools-9.4.0~/modules/Makefile.in open-vm-tools-9.4.0/modules/Makefile.in
--- open-vm-tools-9.4.0~/modules/Makefile.in	2013-09-23 08:51:48.000000000 -0700
+++ open-vm-tools-9.4.0/modules/Makefile.in	2013-11-01 15:48:53.906211120 -0700
@@ -595,7 +595,7 @@
 @LINUX_TRUE@export vmsyncdir := $(MODULES_DIR)/drivers/misc
 @LINUX_TRUE@export vmxnetdir := $(MODULES_DIR)/drivers/net
 @LINUX_TRUE@export vsockdir := $(MODULES_DIR)/net/vsock
-@LINUX_TRUE@@WITH_ROOT_PRIVILEGES_TRUE@export DEPMOD := depmod -a $(KERNEL_RELEASE)
+@LINUX_TRUE@@WITH_ROOT_PRIVILEGES_TRUE@export DEPMOD := depmod -a $(KERNEL_RELEASE) -b '$(MODULES_DIR)/../../../../'
 
 @SOLARIS_TRUE@export vmhgfsdir := $(MODULES_DIR)
 
