--- bzip2-1.0.6/Makefile-libbz2_so	2015-07-06 17:21:34.000000000 -0700
+++ bzip2-1.0.6/Makefile-libbz2_so	2015-07-06 17:29:43.000000000 -0700
@@ -34,8 +34,14 @@
       decompress.o \
       bzlib.o
 
+ifeq ($(shell uname -s), Darwin)
+  SONAME=install_name
+else
+  SONAME=soname
+endif
+
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
+	$(CC) -shared -Wl,-$(SONAME) -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.6 libbz2.so.1.0
