diff -urN open-vm-tools-9.2.2/modules/linux/shared/compat_mm.h open-vm-tools-9.2.2-prev/modules/linux/shared/compat_mm.h
--- open-vm-tools-9.2.2/modules/linux/shared/compat_mm.h	2013-03-07 17:10:43.230963434 -0800
+++ open-vm-tools-9.2.2-prev/modules/linux/shared/compat_mm.h	2013-03-07 17:10:09.032974789 -0800
@@ -99,18 +99,8 @@
    vmtruncate(inode, size);                                                   \
    result;                                                                    \
 })
-#elif LINUX_VERSION_CODE < KERNEL_VERSION(3, 8, 0)
-#define compat_vmtruncate(inode, size) vmtruncate(inode, size)
 #else
-#define compat_vmtruncate(inode, size)                                        \
-({                                                                            \
-   result = inode_newsize_ok(inode, size);                                    \
-   if (!result)                                                               \
-   {                                                                          \
-      truncate_setsize(inode, size);                                          \
-   }                                                                          \
-   result;                                                                    \
-})
+#define compat_vmtruncate(inode, size) vmtruncate(inode, size)
 #endif
 
 
