--- open-vm-tools-9.2.2/configure	2013-01-05 02:15:45.127000018 -0800
+++ open-vm-tools-9.2.2/configure	2013-01-05 02:23:17.204000342 -0800
@@ -22203,7 +22203,7 @@
 
 ### General flags / actions
 CFLAGS="$CFLAGS -Wall"
-CFLAGS="$CFLAGS -Werror"
+# CFLAGS="$CFLAGS -Werror"
 
 # -Wno-unknown-pragmas is due to gcc not understanding '#pragma ident' in Xlib.h on OpenSolaris.
 for TEST_CFLAG in -Wno-pointer-sign -Wno-unused-value -fno-strict-aliasing \
--- open-vm-tools-9.2.2/configure.ac	2013-01-05 02:15:45.139000016 -0800
+++ open-vm-tools-9.2.2/configure.ac	2013-01-05 02:22:25.346000166 -0800
@@ -832,7 +832,7 @@
 
 ### General flags / actions
 CFLAGS="$CFLAGS -Wall"
-CFLAGS="$CFLAGS -Werror"
+# CFLAGS="$CFLAGS -Werror"
 
 # -Wno-unknown-pragmas is due to gcc not understanding '#pragma ident' in Xlib.h on OpenSolaris.
 for TEST_CFLAG in -Wno-pointer-sign -Wno-unused-value -fno-strict-aliasing \
